clio-client/src/main/scala/org/broadinstitute/clio/client/commands/Commands.scala

```diff
@@ -1,30 +1,84 @@
 package org.broadinstitute.clio.client.commands
 
+import java.time.OffsetDateTime
+
 import akka.http.scaladsl.model.HttpResponse
-import enumeratum._
-import org.broadinstitute.clio.client.parser.BaseArgs
+import caseapp.{AppName, ProgName, Recurse}
+import caseapp.core.ArgParser
+import io.circe.parser.parse
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.client.webclient.ClientAutoDerivation._
 import org.broadinstitute.clio.client.webclient.ClioWebClient
+import org.broadinstitute.clio.transfer.model.{
+  TransferWgsUbamV1Key,
+  TransferWgsUbamV1Metadata,
+  TransferWgsUbamV1QueryInput
+}
+import org.broadinstitute.clio.util.model.{DocumentStatus, Location}
 
-import scala.collection.immutable.IndexedSeq
 import scala.concurrent.{ExecutionContext, Future}
 
-sealed trait CommandType extends EnumEntry
-
-object Commands extends Enum[CommandType] {
-  override val values: IndexedSeq[CommandType] = findValues
+object CustomArgParsers {
+  implicit def locationParser: ArgParser[Location] = {
+    ArgParser.instance[Location]("location") { location =>
+      Right(Location.pathMatcher(location))
+    }
+  }
+  implicit def documentStatusParser: ArgParser[DocumentStatus] = {
+    ArgParser.instance[DocumentStatus]("documentStatus") { documentStatus =>
+      Right(
+        DocumentStatus.namesToValuesMap
+          .getOrElse(documentStatus, DocumentStatus.Normal)
+      )
+    }
+  }
+  implicit def offsetDateTimeParser: ArgParser[OffsetDateTime] = {
+    ArgParser.instance[OffsetDateTime]("date") { offsetDateAndTime =>
+      Right(OffsetDateTime.parse(offsetDateAndTime))
+    }
+  }
+}
 
-  case object AddWgsUbam extends CommandType
-  case object QueryWgsUbam extends CommandType
+@AppName("Clio Client")
+@ProgName("clio-client")
+final case class CommonOptions(bearerToken: String = "")
 
-  val pathMatcher = Map(
-    AddWgsUbam.toString -> AddWgsUbam,
-    QueryWgsUbam.toString -> QueryWgsUbam
-  )
+sealed abstract class Command extends Product with Serializable {
+  def execute(webClient: ClioWebClient, bearerToken: String)(
```

Comment:
I'd still prefer to keep the `execute` definitions separated from the `Command` instance definitions, to avoid being forced by `sealed` to put all the implementations in this file.